### PR TITLE
[API] Add LLaMA attention API.

### PR DESCRIPTION
**Pull request by changqi1**
```bash
$ ./ut/layers_attention_test
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from AttentionLLaMA
[ RUN      ] AttentionLLaMA.bfloat16_t
>> create context: 4096 128
>> create llama_attention_key: 0x7f8fe6804040_0x7f8fe6808040_0x7f8fe6808040_0x7f8fe2803040_1_128_32_32
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.153258 sec
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.004665 sec
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.001864 sec
[       OK ] AttentionLLaMA.bfloat16_t (755 ms)
[ RUN      ] AttentionLLaMA.float16_t
>> create context: 4096 128
>> create llama_attention_key: 0x7f8fe6804040_0x7f8fe6808040_0x7f8fe6808040_0x7f8fe2803040_2_128_32_32
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.119574 sec
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.046373 sec
[ RUNTIME  ] XFT::invokeAttentionLLaMA 0.039210 sec
[       OK ] AttentionLLaMA.float16_t (1601 ms)
[----------] 2 tests from AttentionLLaMA (2356 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test case ran. (2357 ms total)
[  PASSED  ] 2 tests.

```

### CODE REVIEW

File: tests/ut/layers_attention_test.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2023 Intel Corporation
```
Comment (Duyi-Wang) [inline tests/ut/layers_attention_test.cpp:1]: 2024

File: src/models/model_factory.h [hunk 2]
```diff
@@ -109,4 +109,4 @@ class DecoderRegister {
     MODEL(IMPLEMENT, CLASS, NAME)
 
 #define REGISTER_MODEL(CLASS, NAME) \
-    MODEL(REGISTER, CLASS, NAME)
\ No newline at end of file
+    MODEL(REGISTER, CLASS, NAME)
```
[the remark this answers is not in the document]
Comment (changqi1) [reply src/models/model_factory.h:112]: I have added a blank line.

File: src/layers/attention.cpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        }
+
+        ctx->resize(batchSize, inputSeqLen, pastSeqLen);
+        hpj::Matrix<float> actBuffers;
+        actBuffers.Resize(batchSize * inputSeqLen * 2, hiddenSize);
+        float *attnMask = prepareAttnMask(ctx, step);
+
+        int workers = 1;
+        int headsPerSplit = (ctx->kvHeadNum + workers - 1) / workers;
+        kvCacheMgr->resize(maxPositions, batchSize, headsPerSplit, attHeadDim);
```
[the remark this answers is not in the document]
Comment (changqi1) [reply src/layers/attention.cpp:170]: Yes, KV cache is managed by us.

File: include/layers_attention.h [hunk 4]
```diff
@@ -44,4 +58,11 @@ void invokeAttention(DataType dt,
         const int batch_size, const int *token_lens, const void *kcache, const void *vcache, int *kvcache_shape,
         int *block_tables, int *block_nums, int *context_lens, int layer_id, bool is_prefill, int *slot_mapping);
 
+void invokeAttentionLLaMA(DataType dt, int batchSize, int inputSeqLen, int attHeadDim, int attHeadNum, int kvHeadNum,
```
[the remark this answers is not in the document]
Comment (changqi1) [reply include/layers_attention.h:61]: The maxSeqLength is only from Qwen, others are our needs.

[the remark this answers is not in the document]
Comment (changqi1) [reply include/layers_attention.h:61]: Done

File: tests/ut/layers_attention_test.cpp (hunk 1 above)
Comment (changqi1) [reply tests/ut/layers_attention_test.cpp]: Done

**Comment by changqi1** [comment]
Waiting to add rmsnorm/layernorm param.

**Comment by changqi1** [comment]
@pujiang2018  @Duyi-Wang  Done. 
Attention APIs don't have the LayerNorm function.
The LayerNorm function will add in Decoder Layer API.

**Review by pujiang2018 — APPROVED** [review]

**Pull request merged**